subsys/net/openthread/rpc/server/ot_rpc_udp_server.c

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <nrf_rpc/nrf_rpc_serialize.h>
+#include <ot_rpc_ids.h>
+#include <ot_rpc_types.h>
+#include <ot_rpc_common.h>
+
+#include <nrf_rpc_cbor.h>
+
+#include <openthread/error.h>
+#include <openthread/instance.h>
+
+#include <openthread/udp.h>
+
+#include <zephyr/net/openthread.h>
+#include <zcbor_decode.h>
+#include <string.h>
+
+#include "ot_rpc_message_server.h"
+
+#define OT_UDP_SOCKETS 8
+
+typedef struct nrfUdpSocket {
+	ot_socket_key mKey;
+	otUdpSocket mSocket;
+} nrfUdpSocket;
```

Comment:
Again incompatible style.
Why do you need separate `type`?

Reply:
Coding style changed.

As for the new type. on this site we need registry with preallocated sockets. Unlike ot message otUdpSocket is also a element in list (ptr to next is embedded into this  data type). This means that I have to leave all fields intact. As for the key it's actually a pointer to same structure on client site. Instead of creating similar registry I'm keeping the e callback and context data in this struct.